index.html

```diff
@@ -1787,6 +1935,296 @@ <h2>
+              </li>
+              <li>|item|["accept"] doesn't [=map/exist=].
+              </li>
+              <li>|item|["accept"] is not a [=dictionary=].
+              </li>
+              <li>|item|["accept"] is [=list/empty].
+              </li>
+            </ul>
+          </li>
+          <li>Let |url:URL| be the result of [=URL Parser|parsing=]
+          |item|["action"] with |start_url URL| as the base URL.
+          </li>
+          <li>If |url| is failure, return failure.
+          </li>
+          <li>If |url| is not [=manifest/within scope=] of |scope|, return
+          failure.
+          </li>
+          <li>Let |accept:ordered map| be an [=ordered map=].
+          </li>
+          <li>[=map/iterate=] |mime_type_string:string| → |extensions| of
+          |item|["accept"]
+            <ol>
+              <li>If |extensions| is not a [=string=] or [=list=], return
+              failure.
+              </li>
+              <li>If |extensions| is the empty string or an [=list/empty=]
+              [=list=], return failure.
+              </li>
+              <li>If |extensions| is a [=list=] with items that are not
+              [=string=]s, return failure.
+              </li>
+              <li>Let |mime_type_parsed:mime type| be the result of running the
+              steps of [=parse a mime type=] on |mime_type_string|.
+              </li>
+              <li>If |mime_type_parsed|.type is not listed in as a top-level
+              type in [[IANA_MEDIA_TYPES]], return failure.
+              </li>
+              <li>Add |mime_type_string| → |extensions| to |accept|.
+              </li>
+            </ol>
+          </li>
+          <li>Let |file_handler:ordered map| be |ordered map| «[ "action" →
+          |url|, "name" → |item|["name"], "accept" → |accept| ]».
+          </li>
+          <li>[=Process image resources=] passing |item|["icons"],
+          |file_handler|, |manifest URL|, and "icons".
+          </li>
+          <li>The user agent MAY replace |file_handler|["icons"] with the
+          application icon.
+          </li>
+          <li>If |file_handler|["name"] doesn't [=map/exist=], the user agent
+          MAY populate a suitable name to describe the file type.
+          </li>
+          <li>Return |file_handler|.
+          </li>
+        </ol>
+      </section>
+    </section>
+    <section data-cite="file-system-access">
+      <h2>
+        <dfn>Execute a file handler launch</dfn>
+      </h2>
+      <p>
+        A {{Window/window}} has an associated {{LaunchConsumer}} <dfn>assigned
+        launch consumer</dfn>.
+      </p>
+      <p>
+        A {{Window/window}} has an associated [=list=] of {{LaunchParams}}
+        <dfn>unconsumed launch params</dfn>.
+      </p>
+      <p>
+        When a [=file type=] is registered with a web app and one or more of
+        these registered files are launched on the platform, run the following
```

Comment:
these steps are also run after every same-origin page navigation (this supports refreshing the page or redirecting)

Reply:
Hm.... @inexorabletash or @marcoscaceres I need some spec help here - how do you think I should specify this? do the above "assigned launch consumer" and "unconsumed launch params" live across same-origin redirects?

Reply:
I think you'll [browsing session](https://html.spec.whatwg.org/multipage/history.html#browsing-sessions) is where you'll need to have the launch params live (rather than window), and then a window has access to the params via its browsing session. 

I have low confidence in my answer here, maybe @jakearchibald can share some wisdom?

Reply:
After some more thought, I think this makes more sense if it's based on the app scope. That is, any navigation or reload that remains within app scope gets the file handles. (Originally I was thinking of origin because it's the standard security boundary, but if we're supporting scope extensions then I suppose the security boundary ought to be a set of origins.)

Reply:
@jakearchibald do you have thoughts?

@alancutter does the existing spec here generally work for you other than the navigation scope part?

Reply:
I wrote a small [doc](https://docs.google.com/document/d/1EeweceXvS55VLzj-YYgUPIa8sq3vHnYEcy5yvoYX54A/edit) for removing the re-enqueuing on navigation implementation currently in Chrome.

The LaunchQueue/LaunchParams API surface LGTM otherwise.